packages/beacon-node/src/network/reqresp/handlers/index.ts

```diff
@@ -13,48 +14,54 @@ import {onLightClientUpdatesByRange} from "./lightClientUpdatesByRange.js";
 import {onStatus} from "./status.js";
 
 export interface ReqRespHandlers {
-  onStatus: HandlerTypeFromMessage<typeof protocols.Status>;
-  onBeaconBlocksByRange: HandlerTypeFromMessage<typeof protocols.BeaconBlocksByRange>;
-  onBeaconBlocksByRoot: HandlerTypeFromMessage<typeof protocols.BeaconBlocksByRoot>;
-  onBeaconBlockAndBlobsSidecarByRoot: HandlerTypeFromMessage<typeof protocols.BeaconBlockAndBlobsSidecarByRoot>;
-  onBlobsSidecarsByRange: HandlerTypeFromMessage<typeof protocols.BlobsSidecarsByRange>;
-  onLightClientBootstrap: HandlerTypeFromMessage<typeof protocols.LightClientBootstrap>;
-  onLightClientUpdatesByRange: HandlerTypeFromMessage<typeof protocols.LightClientUpdatesByRange>;
-  onLightClientFinalityUpdate: HandlerTypeFromMessage<typeof protocols.LightClientFinalityUpdate>;
-  onLightClientOptimisticUpdate: HandlerTypeFromMessage<typeof protocols.LightClientOptimisticUpdate>;
+  onStatus: (req: phase0.Status, peerId: PeerId) => AsyncIterable<ResponseOutgoing>;
+  onBeaconBlocksByRange: ProtocolHandler;
+  onBeaconBlocksByRoot: ProtocolHandler;
+  onBeaconBlockAndBlobsSidecarByRoot: ProtocolHandler;
+  onBlobsSidecarsByRange: ProtocolHandler;
+  onLightClientBootstrap: ProtocolHandler;
+  onLightClientUpdatesByRange: ProtocolHandler;
+  onLightClientFinalityUpdate: ProtocolHandler;
+  onLightClientOptimisticUpdate: ProtocolHandler;
 }
 /**
  * The ReqRespHandler module handles app-level requests / responses from other peers,
  * fetching state from the chain and database as needed.
  */
 export function getReqRespHandlers({db, chain}: {db: IBeaconDb; chain: IBeaconChain}): ReqRespHandlers {
   return {
-    async *onStatus(protocol) {
-      yield* onStatus(protocol, chain);
+    async *onStatus() {
+      yield* onStatus(chain);
     },
-    async *onBeaconBlocksByRange(protocol, req) {
-      yield* onBeaconBlocksByRange(protocol, req, chain, db);
+    async *onBeaconBlocksByRange(req) {
+      const body = ssz.phase0.BeaconBlocksByRangeRequest.deserialize(req.data);
```

Comment:
same to previous comment, why do we have to deserialize each protocol separately here instead of in `handleRequest()` method of reqresp package?

Reply:
I think its because of the desired architecture here: https://github.com/ChainSafe/lodestar/issues/5378#issuecomment-1528125740

Handlers will be in the main thread and the `handleRequest` will be in the network thread.

Reply:
thanks for clarifying @wemeetagain , that makes a lot of sense 👍